atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+                        continue
+                    self._observable_info[observable_type][observable_name][observable_label][key] = observable_attribute
+        return self._observable_info
+
+    def get_atomsgroup_fromh5md(self, nomad_sec, h5md_sec_particlesgroup):
+        for i_key, key in enumerate(h5md_sec_particlesgroup.keys()):
+            particles_group = {group_key: self._data_parser.hdf5_getter(h5md_sec_particlesgroup[key], group_key) for group_key in h5md_sec_particlesgroup[key].keys()}
+            sec_atomsgroup = nomad_sec.m_create(AtomsGroup)
+            sec_atomsgroup.type = particles_group.pop('type', None)
+            sec_atomsgroup.index = i_key
+            sec_atomsgroup.atom_indices = particles_group.pop('indices', None)
+            sec_atomsgroup.n_atoms = len(sec_atomsgroup.atom_indices) if sec_atomsgroup.atom_indices is not None else None
+            sec_atomsgroup.is_molecule = particles_group.pop('is_molecule', None)
+            sec_atomsgroup.label = particles_group.pop('label', None)
+            sec_atomsgroup.composition_formula = particles_group.pop('formula', None)
+            particles_subgroup = particles_group.pop('particles_group', None)
+            # set the remaining attributes
+            for particles_group_key in particles_group.keys():
+                unit = None
+                val = particles_group.get(particles_group_key)
+                has_units = hasattr(val, 'units') if val is not None else None
+                if has_units:
+                    unit = val.units
+                    val = val.magnitude
+                sec_atomsgroup.x_h5md_parameters.append(ParamEntry(kind=particles_group_key, value=val, unit=unit))
+            # get the next atomsgroup
+            if particles_subgroup:
+                self.get_atomsgroup_fromh5md(sec_atomsgroup, particles_subgroup)
+
+    def check_metainfo_for_key_and_Enum(self, metainfo_class, key, val):
+        if key in metainfo_class.__dict__.keys():
+            quant = metainfo_class.__dict__.get(key)
+            if quant.get('type') is not None:
+                if type(quant.type).__name__ == 'MEnum':
+                    if val in quant.type._list:
+                        return key
+                    else:
+                        return 'x_h5md_' + key
+                else:
+                    return key
+            else:
+                self.logger.warning(key + 'in ' + metainfo_class + ' is not a Quantity or does not have an associated type.')  # Not sure if this can ever happen
+                return key
+        else:
+            return 'x_h5md_' + key
+
+    @property
+    def parameter_info(self):
+        if self._parameter_info is None:
+            parameters_group = self.h5md_groups.get('parameters')
+            if parameters_group is None:
+                return
+            self._parameter_info = {
```

Comment:
i think you return this if parameters_group is None, otherwise you will always do this whenever you access paramter_info.

Reply:
Can I just return {}? I can't populate these subsections in any case if there is no parameter info

Reply:
nm, done.